packages/jsii-rosetta/lib/commands/extract.ts

```diff
@@ -49,16 +63,19 @@ export async function extractSnippets(
   const translatorOptions: RosettaTranslatorOptions = {
     includeCompilerDiagnostics: options.includeCompilerDiagnostics,
     assemblies: assemblies.map((a) => a.assembly),
+    tablet: options.append ? await LanguageTablet.fromOptionalFile(options.outputFile) : undefined,
   };
 
   const translator = options.translatorFactory
     ? options.translatorFactory(translatorOptions)
     : new RosettaTranslator(translatorOptions);
 
-  if (options.cacheTabletFile) {
-    await translator.loadCache(options.cacheTabletFile);
+  // Choose the cache if provided, or the output file if we plan to append to it.
+  const cacheTabletFile = options.cacheTabletFile ?? (options.append ? options.outputFile : undefined);
```

Comment:
We need to support BOTH the append-file and the cache file as sources of existing snippets.